### ISSUE TITLE
Investigate if there's a clean way to implement resizable video

### ISSUE THREAD

**Issue by caprica** [caprica/vlcj-javafx#8]
The way LibVLC works is on video playback to callback the application to get the buffer size (fixed width and height).

If the video window is resized in the client, it is stuck with the old buffer.

Is there a way to dynamically resize the buffer without stopping/restarting the video.

**Comment by caprica** [comment]
Maybe instead of using format callbacks it's possible to invoke libvlc_video_set_format while the video is playing, although LibVLC API documentation reports that this method is bugged for certain colourspaces.

**Comment by caprica** [comment]
After further testing, the format callbacks (the recommended way to set video format) can not be changed after the video has started playing.

**Comment by encircled** [comment]
I have workaround for this issue using WritableImage for displaying video and fittingToSize for resizing. This resizing is smooth on my laptop. I can create a pull request if u want

**Comment by caprica** [comment]
I would be very happy to see your pull request.

The lack of a good way to do resizing, so far, is the biggest problem IMHO with vlcj on JavaFX.

**Comment by encircled** [comment]
Sure, will create it tomorrow.
Well, the biggest problem that I faced is when I open video > 10gb it takes about 10 sec before playing is get started. The strange thing is, that "display" method on DirectMediaPlayerComponent is called, but canvas wont display it... The same video in HeadlessPlayer is opened in about 1 sec

**Comment by caprica** [comment]
Your problem with 10gb video, is that related to your resizing solution or is it a different problem?

Using the DirectMediaPlayerComponentTest my 8gb sample plays immediately.

Using the AnimationTimerJavaFXDirectRenderingTest I do see a notable delay.

Very odd.

**Comment by encircled** [comment]
No, it's not related to resizing or vlcj, something about javaFX, so never mind.

**Comment by caprica** [comment]
Fixed by new sample at 3f868f3c4e53dca7ee8f015f189ba75700655d8d.

The idea is simply to use a native video buffer format in the maximum possible resolution, and then scale it down on resize. This is probably the best way to do it.

**Comment by caprica** [comment]
This other unrelated issue about the long time taken to open a very large video file, I wonder if VLC might be indexing your file before it plays it - the file I used as a test did indeed lack an index. I wonder if somehow that is much worse when using "direct" media players.

**Comment by encircled** [comment]
I have created a simple single-window fx app with direct player. It plays a large video with a tiny delay. Then I copy-pasted it into my project, run only that app (+ project things like maven dependencies etc) and opening the same video file again takes very long... Also, I tried to copy all project-specific dependencies from my project to the test app and it din't cause any changes. Both projects are located on the same hard drive and I run it the same way from Intellij and/or maven.

Generally: when I open it from my project java heap jumps from ~50mb to ~1Gb of ram and delay is is caused by memory allocation. When I open it from test app - it uses about 300mb. Profiler says, that it is "int[]", which is used as a VLC buffer.

**Comment by caprica** [comment]
I dunno, I just ran my 8gb video file in a stand-alone application and it started immediately.

What I remember from the last time I did profiling on this was PixelWriter generates massive amounts of garbage.

The video buffer that vlcj provides for VLC to render into is supposed to be allocated once and re-user over and again.

If you disable your pixel writer (but let VLC keep filling the native buffer) you will see some interesting behaviour I think.
